cpp/include/cuml/cluster/dbscan_api.h

```diff
@@ -17,10 +17,6 @@
 
 #include <cuml/cuml_api.h>
 
-#ifdef __cplusplus
```

Comment:
Any reason why this ifdef is removed?

Reply:
This `#ifdef` was removed because the some of the C++ and C-API functions used to be defined in the same file (see `knn.cu`) and would include both the C++ and C-API headers. In this PR I separated all C-API definitions and declarations into `*_api.h/cpp` files which removed the need for `#ifdef __cplusplus`.

I think this better separates the two different compilations and cleans up include tree since C++ files only include C++ headers and vice-versa.

Looking now, `svm_api.h`, `glm_api.h` and `holtwinters_api.h` also have these `#ifdefs` which could be removed as well.

Reply:
This `#ifdef` was removed because the some of the C++ and C-API functions used to be defined in the same file (see `knn.cu`) and would include both the C++ and C-API headers. In this PR I separated all C-API definitions and declarations into `*_api.h/cpp` files which removed the need for `#ifdef __cplusplus`.

I did accidentally remove the `extern "C"` in other places and fixed that up as a result of this comment. All of the C-API files now follow the same pattern.

Reply:
This header file only contains the 2 dbscan C-API methods. So, shouldn't this header file also be wrapped with the `__cplusplus` macro guard?

Reply:
I took this into consideration and updated the design to revert these changes. See this comment for more info: https://github.com/rapidsai/cuml/pull/3402#issuecomment-772092988